inline comment closure removed

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**
* Windows 7
* **ESLint Version: v3.10.1**
* **Node Version: v4.4.7**
* **npm Version: 3.10.3**

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<!-- Paste your configuration below: -->
```
{
  "env": {
    "browser": true,
    "amd": true
  },
  "globals": {},
  "rules": {
    "semi": [
      "error",
      "always"
    ],
    "no-bitwise": 0,
    "no-cond-assign": 0,
    "camelcase": 2,
    "curly": 2,
    "no-debugger": 0,
    "eqeqeq": 2,
    "no-eq-null": 2,
    "no-eval": 0,
    "no-unused-expressions": 2,
    "guard-for-in": 0,
    "block-scoped-var": 2,
    "strict": 0,
    "comma-dangle": [
      "error",
      "never"
    ],
    "lines-around-directive": [
      "error",
      {
        "before": "never",
        "after": "always"
      }
    ],
    "object-curly-newline": [
      "error",
      "always"
    ],
    "wrap-iife": [
      2,
      "any"
    ],
    "keyword-spacing": [
      "error",
      {
        "before": true
      }
    ],
    "key-spacing": [
      "error",
      {
        "beforeColon": false
      }
    ],
    "no-iterator": 0,
    "no-use-before-define": 0,
    "comma-style": [
      2,
      "last"
    ],
    "space-before-blocks": [
      2,
      "always"
    ],
    "space-before-function-paren": [
      2,
      "always"
    ],
    "no-trailing-spaces": [
      2,
      {
        "skipBlankLines": true
      }
    ],
    "space-infix-ops": [
      2,
      {
        "int32Hint": false
      }
    ],
    "no-mixed-spaces-and-tabs": 2,
    "no-loop-func": 2,
    "no-multi-str": 0,
    "new-cap": 2,
    "no-caller": 2,
    "no-empty": 0,
    "no-new": 2,
    "no-plusplus": 0,
    "no-proto": 0,
    "no-script-url": 2,
    "no-shadow": 2,
    "dot-notation": 2,
    "no-new-func": 0,
    "no-new-wrappers": 0,
    "no-undef": 2,
    "no-unused-vars": 2,
    "no-invalid-this": 0,
    "no-with": 0,
    "max-len": [
      2,
      {
        "code": 120,
        "ignoreComments": true
      }
    ],
    "indent": [
      "error",
      "tab"
    ],
    "quotes": [
      "error",
      "double",
      {
        "avoidEscape": true
      }
    ],
    "quote-props": [
      2,
      "as-needed",
      {
        "unnecessary": true
      }
    ],
    "complexity": [
      2,
      10
    ]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

<!-- Paste the source code below: -->
```js
function test(argument) {

	return register("baz", [foo, bar, baz],
			/** dont ask me why people do this **/{
			baseClass: "bar"
			});
};
```

**What did you expect to happen?**

The comment closure not to be removed

```js
function test (argument) {

	return register("baz", [foo, bar, baz],
			/** dont ask me why people do this **/		{
			baseClass: "bar"
		});
};
```

**What actually happened? Please include the actual, raw output from ESLint.**

comment closure has been removed

```sh
  79:13  error  Parsing error: Unexpected token :

✖ 1 problem (1 error, 0 warnings)
```
```js
function test (argument) {

	return register("baz", [foo, bar, baz],
			/** dont ask me why people do this 		{
			baseClass: "bar"
		});
};
```
**EDIT** updated to ESLint v3.10.1

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -255,20 +255,18 @@ module.exports = {
          * @param {int} lastNodeCheckEndOffset Number of charecters to skip from the end
          * @returns {void}
          */
-        function report(node, needed, gottenSpaces, gottenTabs, loc, isLastNodeCheck, lastNodeCheckEndOffset) {
+        function report(node, needed, gottenSpaces, gottenTabs, loc, isLastNodeCheck) {
             if (gottenSpaces && gottenTabs) {
 
                 // To avoid conflicts with `no-mixed-spaces-and-tabs`, don't report lines that have both spaces and tabs.
                 return;
             }
 
-            lastNodeCheckEndOffset = lastNodeCheckEndOffset || 0;
-
             const desiredIndent = (indentType === "space" ? " " : "\t").repeat(needed);
 
             const textRange = isLastNodeCheck
-                ? [node.range[1] - gottenSpaces - gottenTabs - 1 - lastNodeCheckEndOffset, node.range[1] - 1 - lastNodeCheckEndOffset]
-                : [node.range[0] - gottenSpaces - gottenTabs, node.range[0]];
+                ? [node.range[1] - node.loc.end.column, node.range[1] - node.loc.end.column + gottenSpaces + gottenTabs]
+                : [node.range[0] - node.loc.start.column, node.range[0] - node.loc.start.column + gottenSpaces + gottenTabs];
 
             context.report({
                 node,
@@ -406,13 +404,11 @@ module.exports = {
          */
         function checkLastReturnStatementLineIndent(node, firstLineIndent) {
             const nodeLastToken = sourceCode.getLastToken(node);
-            let lastNodeCheckEndOffset = 0;
             let lastToken = nodeLastToken;
 
             // in case if return statement ends with ');' we have traverse back to ')'
             // otherwise we'll measure indent for ';' and replace ')'
             while (lastToken.value !== ")") {
-                lastNodeCheckEndOffset++;
                 lastToken = sourceCode.getTokenBefore(lastToken);
             }
 
@@ -433,8 +429,7 @@ module.exports = {
                     endIndent.space,
                     endIndent.tab,
                     { line: lastToken.loc.start.line, column: lastToken.loc.start.column },
-                    true,
-                    lastNodeCheckEndOffset
+                    true
                 );
             }
         }

--- a/tests/fixtures/rules/indent/indent-valid-fixture-1.js
+++ b/tests/fixtures/rules/indent/indent-valid-fixture-1.js
@@ -43,7 +43,7 @@ if (a) {
 /**/var b; // NO ERROR: single line multi-line comments followed by code is OK
 /*
  *
- */var b; // ERROR: multi-line comments followed by code is not OK
+*/ var b; // ERROR: multi-line comments followed by code is not OK
 
 var arr = [
   a,

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -3504,6 +3504,32 @@ ruleTester.run("indent", rule, {
             ")",
             parserOptions: {ecmaFeatures: {globalReturn: true}},
             errors: expectedErrors([3, 0, 4, "ReturnStatement"])
+        },
+
+        // https://github.com/eslint/eslint/issues/7604
+        {
+            code:
+            "if (foo) {\n" +
+            "        /* comment */bar();\n" +
+            "}",
+            output:
+            "if (foo) {\n" +
+            "    /* comment */bar();\n" +
+            "}",
+            errors: expectedErrors([2, 4, 8, "ExpressionStatement"])
+        },
+        {
+            code:
+            "foo('bar',\n" +
+            "        /** comment */{\n" +
+            "        ok: true" +
+            "    });",
+            output:
+            "foo('bar',\n" +
+            "    /** comment */{\n" +
+            "        ok: true" +
+            "    });",
+            errors: expectedErrors([2, 4, 8, "ObjectExpression"])
         }
     ]
 });

